lib/cfg.py

```diff
@@ -251,6 +253,83 @@ def cfg_version(filename):
         cfg[section] = d
     return cfg
 
+def cfg_altiumWorkspace(options):
+    """
+    ricerca del nome di tutti i progetti all'interno del file Workspace
+    """
+    risultato=True
+    path_dict = {}
+    config = ConfigParser.RawConfigParser()
+    config.read(options.ws)
+
+    for i in config.sections():
+        try:
+            temp = config.get(i, 'ProjectPath')
+            temp = temp.split('/')
+            k = []
+            p = []
+
+            if len(temp) > 1:
+                k = os.path.splitext(temp[1])[0]
+                p = os.path.join(*temp)
+            else:
+                k = os.path.splitext(temp[0])[0]
+                p = os.path.join("./", *temp)
+
+            if len(k) == 0 or len(p) == 0:
+                lib.error("Errore nel parsing del workspace: %s" % e,
+                            self.handler, terminal=self.terminal)
+                risultato =False
+
+            k = k.lower()
+            path_dict[k] = p
+
+        except ConfigParser.NoOptionError:
+            pass
+    path=''
+    ws=options.ws.split('/')
+    for i in range(0, (len(ws)-1)):
+        path=os.path.join(path, ws[i])
+
+    """
+    ricerca parametri per ogni progetto e creazione dizionario con {parametro: valore}
+    ricerca e verifica esistenza file a cui fare il mergebom e messi in una lista
+    """
+    parametri_dict={}
+    file_BOM=[]
+    if risultato:
```

Comment:
Come prima cosa, mi viene da pensare che se `risultato` è `False`, non puoi fare altro, e quindi puoi ritornare prima, con un valore di ritorno che si capisce che c'è stato un errore. In questo modo il codice è più semplice e si legge meglio.